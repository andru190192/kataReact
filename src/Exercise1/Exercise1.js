import React, {Component} from 'react';

export class Exercise1 extends Component {
  constructor(props) {
    super(props);
    this.state = {
      alphabet: 'abcdefghijklmnopqrstuvwxyz'
    };
  }
  
  render() {
    const {alphabet} = this.state;

    return (
      <div className="container">
        Alphabet
        <ul>
          {
            alphabet.split('').map((element, i) => 
              <li key={i}>
                {`${element}${alphabet}`.substring(i + 1, alphabet.length + 1) + `${alphabet}`.substring(0, i)}
              </li>
            )
          }
        </ul>
      </div>
    );
  }
}

export default Exercise1;
