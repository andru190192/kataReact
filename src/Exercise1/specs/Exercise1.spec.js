import React from 'react';
import {shallow} from 'enzyme';
import Exercise1 from '../Exercise1';

describe('Exercise1', () => {
  describe('Exercise1', () => {
    it('should print 26 li', () => {
      const wrapper = shallow(
        <Exercise1 />,
      );
      const li = wrapper.find('li');
      expect(li).toHaveLength(26);
    });
    it('should print 26 letters for line', () => {
      const wrapper = shallow(
        <Exercise1 />,
      );
      const li = wrapper.find('li');
      expect(li.length).toEqual(26);
    });
  });
});
