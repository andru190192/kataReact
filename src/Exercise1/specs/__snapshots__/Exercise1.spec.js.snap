// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exercise1 Exercise1 should toMatchSnapshot 1`] = `
<div
  className="container"
>
  Abecedario
  <ul>
    <span
      key="0"
    >
      a
    </span>
    <span
      key="1"
    >
      b
    </span>
    <span
      key="2"
    >
      c
    </span>
    <span
      key="3"
    >
      d
    </span>
    <span
      key="4"
    >
      e
    </span>
    <span
      key="5"
    >
      f
    </span>
    <span
      key="6"
    >
      g
    </span>
    <span
      key="7"
    >
      h
    </span>
    <span
      key="8"
    >
      i
    </span>
    <span
      key="9"
    >
      j
    </span>
    <span
      key="10"
    >
      k
    </span>
    <span
      key="11"
    >
      l
    </span>
    <span
      key="12"
    >
      m
    </span>
    <span
      key="13"
    >
      n
    </span>
    <span
      key="14"
    >
      o
    </span>
    <span
      key="15"
    >
      p
    </span>
    <span
      key="16"
    >
      q
    </span>
    <span
      key="17"
    >
      r
    </span>
    <span
      key="18"
    >
      s
    </span>
    <span
      key="19"
    >
      t
    </span>
    <span
      key="20"
    >
      u
    </span>
    <span
      key="21"
    >
      v
    </span>
    <span
      key="22"
    >
      w
    </span>
    <span
      key="23"
    >
      x
    </span>
    <span
      key="24"
    >
      y
    </span>
    <span
      key="25"
    >
      z
    </span>
    }
  </ul>
</div>
`;
